File: services/discount-service/src/main/java/com/fiji/discountservice/controller/DiscountRequest.java
package com.fiji.discountservice.controller;

import jakarta.persistence.Column;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

import java.time.LocalDateTime;

public record DiscountRequest (
     Long id,
     String name, // Название предложения

     String company, // Название предложения

     String description,

     LocalDateTime startDate, // Дата начала предложения

     LocalDateTime endDate, // Дата окончания предложения

     boolean active // Активно ли предложение
)
{}

File: fiji-ui/src/app/services/discount/models/discount-request.ts
export interface DiscountRequest {
  id?: number; // ID может быть необязательным для новых скидок
  name: string; // Название предложения
  company: string; // Компания
  description: string; // Описание
  startDate: string; // Дата начала (в формате ISO 8601)
  endDate: string; // Дата окончания (в формате ISO 8601)
  active: boolean; // Активность
}
